class-components/src/pages/app/App.tsx

```diff
@@ -0,0 +1,81 @@
+import './App.css';
+import Search from '../../components/search/search-component';
+import fetchPokemons from '../../api/api';
+import Spinner from '../../components/spinner/spinner';
+import PokemonList from '../../components/pokemon-list/pokemon-list-component';
+import { useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router';
+import useSearchQuery from '../../hooks/useSearchQuery';
+
+type Pokemon = {
+  id: number;
+  name: string;
+  image: string;
+  description: string;
+};
+
+export default function App() {
+  const [isSpinnerActive, setSpinner] = useState(false);
+  const [pokemonArray, setPokemonArray] = useState<Pokemon[]>([]);
+  const [isErrorActive, setError] = useState(false);
+  const [page, setPage] = useState(1);
+
+  const { pageId } = useParams();
+
+  useEffect(() => {
+    handleClick();
+  }, [pageId]);
+
+  const [, get] = useSearchQuery();
+
+  const handleClick = async () => {
+    console.log("Hello, I'm pokemon");
+    setSpinner(true);
+    try {
+      const allPokemons = await fetchPokemons(get(), Number(pageId));
+      setPage(allPokemons.pages);
+      setPokemonArray(allPokemons.data);
+      setSpinner(false);
+      setError(false);
+    } catch {
+      setError(true);
+      setSpinner(false);
```

Comment:
You might want to handle loading and error states in a finally block to avoid repeating setSpinner/setError in multiple places.